schema.ts

```diff
@@ -1321,11 +1348,79 @@ function checkSchemaPropertyIsRequired<K extends string, L extends string, T ext
   );
 }
 
+/**
+ * Normalizes a schema key into PascalCase.
+ */
 export function normalizeSchemaKey(key: string): string {
   // Colons cause problems in our formula handling.
   return pascalcase(key).replace(/:/g, '_');
 }
 
+/**
+ * Normalizes a schema property key path. This interprets "."s as accessing object properties
+ * and "[]" as accessing array items. Uses normalizeSchemaKey to normalize each part in-between.
+ *
+ * This is used for object schema properties that support path projection.
+ */
+export function normalizeSchemaKeyPath(key: string, normalizedProperties: ObjectSchemaProperties): string {
+  // Try an exact match on the properties first.
+  if (normalizedProperties.hasOwnProperty(normalizeSchemaKey(key))) {
+    return normalizeSchemaKey(key);
+  }
+
+  // Try splitting by . to handle json paths.
+  return key
+    .split('.')
+    .map(val => {
+      let partToNormalize = val;
+      let partToIgnoreNormalization = '';
+
+      // Handles array pathing.
+      if (val.includes('[')) {
+        partToNormalize = val.substring(0, val.indexOf('['));
+        partToIgnoreNormalization = val.substring(val.indexOf('['));
+      }
+
+      return normalizeSchemaKey(partToNormalize) + partToIgnoreNormalization;
+    })
+    .join('.');
+}
+
+/**
+ * Normalizes a schema PropertyIdentifier by converting it to PascalCase.
+ */
+function normalizeSchemaPropertyIdentifier(
+  key: PropertyIdentifier,
+  normalizedProperties: ObjectSchemaProperties,
+): PropertyIdentifier {
+  if (typeof key === 'string') {
+    return normalizeSchemaKeyPath(key, normalizedProperties);
+  }
+
+  const {label, property: value} = key;
+  return {
+    property: normalizeSchemaKeyPath(value, normalizedProperties),
+    label: normalizeSchemaKey(label),
```

Comment:
Is it intentional that we want to normalize this, given that it's just a display label? Before, everything did double-duty as a schema property and a display label, but now we actually have something that's just a display label.

We may decide we want to normalize it if people are using weird casing and we want cards to look consistent. But maybe that's something we could do at render time vs losing the original info at compile time.

Reply:
yeah ill get rid of normalization for now -- was just worried about it appearing inconsistent since we already do some smart "change from pascal case to sentence case" logic at render time